### PR TITLE
SCA Analytics

### PR DESCRIPTION
**Pull request by olgakil**
PR - create analytics log for each SCA scan and pull reqeust

Tests: automation test is added for SCA scan, and manual tests are performed for pull request with SCA scan configuration

- All active GitHub checks for tests, formatting, and security are passing
- The correct base branch (develop) is being used

### CODE REVIEW

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

**Comment by cxflowtestuser** [comment]
### Checkmarx scan completed
[Full Scan Details](http://EC2AMAZ-JP2I9QF/CxWebClient/ViewerMain.aspx?scanid=1010260&projectid=102)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|435
Low|4993
Informational|44

#### Violation Summary
Severity|Count
---|---

#### Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---

File: src/main/java/com/checkmarx/flow/service/GitHubService.java [hunk 1]
```diff
@@ -191,20 +193,69 @@ private void statusExchange(ScanRequest request, HttpEntity<?> httpEntity, Strin
     }
 
     void endBlockMerge(ScanRequest request, ScanResults results, ScanDetails scanDetails) {
+
```
Comment (AvivCX) [inline src/main/java/com/checkmarx/flow/service/GitHubService.java:196]: Can't we take it out of github service and put it in generic layer?

File: src/main/java/com/checkmarx/flow/service/GitHubService.java [hunk 2]
```diff
@@ -191,20 +193,69 @@ private void statusExchange(ScanRequest request, HttpEntity<?> httpEntity, Strin
...
             statusExchange(request, httpEntity, statusApiUrl, "failed to update merge status for completed scan");
         } else {
             log.debug("Pull request blocking is disabled in configuration, no need to unblock.");
+
+            logPullRequestWithSastOsa(request, results, scanDetails);
+        }
+    }
+
+    private void logPullRequestWithSastOsa(ScanRequest request, ScanResults results, ScanDetails scanDetails) {
+
```
Comment (AvivCX) [inline src/main/java/com/checkmarx/flow/service/GitHubService.java:222]: why we call it in the code OSA and not SCA?

**Pull request merged**